packages/mobile/src/screens/favorites-screen/TracksTab.tsx

```diff
@@ -126,11 +129,15 @@ export const TracksTab = () => {
         )
       }
     },
-    [dispatch, isPlaying, playingUid]
+    [playingUid, isPlaying, dispatch, isReachable, isOfflineModeEnabled]
   )
 
   const isLoading = savedTracksStatus !== Status.SUCCESS
-  const hasNoFavorites = savedTracks.entries.length === 0
+  let tracks = savedTracks.entries
+  if (!isReachable && isOfflineModeEnabled) {
+    tracks = Object.values(offlineTracks)
+  }
```

Comment:
nit but I think this would be better as an expression
```suggestion
const tracks =
  !isReachable && isOfflineModeEnabled
    ? Object.values(offlineTracks)
    : savedTracks.entries
```